core/src/thread_parker/unix.rs

```diff
@@ -28,6 +32,12 @@ type tv_nsec_t = libc::c_long;
 
 // Helper type for putting a thread to sleep until some other thread wakes it up
 pub struct ThreadParker {
+    // REVIEW: the safety here seems a bit sketchy mixing threadsafe and
+    // non-threadsafe types. Could the use of each field be documented as to why
+    // it's either a `Cell` or an `UnsafeCell`?
+    //
+    // I think the tl;dr; is that one of these `Cell` shoudl be an
+    // `UnsafeCell`.
```

Comment:
`should_park` is only accessed while holding the mutex, so this should be fine I think? The general rule I've been going with is to put `Copy` types in `Cell` and the rest in `UnsafeCell`.

Reply:
This bit us long ago when dealing with channels at some point, but if interior mutability is used and it's in a threaded context then `UnsafeCell` should be used. Using `Cell` provides a false sense of security since the contract is that a lock must be held *then* the `Cell` can be accessed and that's verify difficult to verify when reading unless there's an `unsafe` block warning "go look for the rationale for safety somewhere"